lib/Parse/ParseSIL.cpp

```diff
@@ -1108,13 +1105,21 @@ bool SILParser::parseSILDottedPathWithoutPound(ValueDecl *&Decl,
                                    SmallVectorImpl<ValueDecl *> &values) {
   // Handle sil-dotted-path.
   Identifier Id;
-  SmallVector<Identifier, 4> FullName;
+  SmallVector<DeclBaseName, 4> FullName;
   SmallVector<SourceLoc, 4> Locs;
   do {
     Locs.push_back(P.Tok.getLoc());
-    if (parseSILIdentifier(Id, diag::expected_sil_constant))
-      return true;
-    FullName.push_back(Id);
+    switch (P.Tok.getKind()) {
+    case tok::kw_subscript:
+      P.consumeToken();
+      FullName.push_back(DeclBaseName::createSubscript());
```

Comment:
@gottesmm or @swiftix, is this correct behavior for finding "subscript" in a textual SIL dotted path? Or is that expected to be treated as a regular identifier? Or can it just not come up?